tests/tests_geomstats/test_landmarks.py

```diff
@@ -1,164 +1,214 @@
 
+import random
+
 import geomstats.backend as gs
-import geomstats.tests
+from geomstats.geometry.euclidean import Euclidean
 from geomstats.geometry.hypersphere import Hypersphere
-from geomstats.geometry.landmarks import Landmarks
-
-
-class TestLandmarks(geomstats.tests.TestCase):
-    def setup_method(self):
-        s2 = Hypersphere(dim=2)
-        r3 = s2.embedding_space
-
-        initial_point = [0.0, 0.0, 1.0]
-        initial_tangent_vec_a = [1.0, 0.0, 0.0]
-        initial_tangent_vec_b = [0.0, 1.0, 0.0]
-        initial_tangent_vec_c = [-1.0, 0.0, 0.0]
-
-        landmarks_a = s2.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec_a
-        )
-        landmarks_b = s2.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec_b
-        )
-        landmarks_c = s2.metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec_c
-        )
-
-        self.n_sampling_points = 10
-        sampling_times = gs.linspace(0.0, 1.0, self.n_sampling_points)
-        landmark_set_a = landmarks_a(sampling_times)
-        landmark_set_b = landmarks_b(sampling_times)
-        landmark_set_c = landmarks_c(sampling_times)
-
-        self.n_landmark_sets = 5
-        self.times = gs.linspace(0.0, 1.0, self.n_landmark_sets)
-        gs.random.seed(1234)
-        self.space_landmarks_in_euclidean_3d = Landmarks(
-            ambient_manifold=r3, k_landmarks=self.n_sampling_points
-        )
-        self.space_landmarks_in_sphere_2d = Landmarks(
-            ambient_manifold=s2, k_landmarks=self.n_sampling_points
-        )
-        self.l2_metric_s2 = self.space_landmarks_in_sphere_2d.metric
-        self.l2_metric_r3 = self.space_landmarks_in_euclidean_3d.metric
-        self.landmarks_a = landmark_set_a
-        self.landmarks_b = landmark_set_b
-        self.landmarks_c = landmark_set_c
-
-    def test_belongs(self):
-        result = self.space_landmarks_in_sphere_2d.belongs(self.landmarks_a)
-        expected = True
-        self.assertAllClose(result, expected)
-
-    def test_belongs_vectorization(self):
-        landmark_sets = gs.array([self.landmarks_a, self.landmarks_b])
-        result = self.space_landmarks_in_sphere_2d.belongs(landmark_sets)
-        expected = gs.array([True, True])
-        self.assertAllClose(result, expected)
-
-    def test_l2_metric_log_and_squared_norm_and_dist(self):
-        """Test that squared norm of logarithm is squared dist."""
-        tangent_vec = self.l2_metric_s2.log(
-            point=self.landmarks_b, base_point=self.landmarks_a
-        )
-        log_ab = tangent_vec
-        result = self.l2_metric_s2.squared_norm(
-            vector=log_ab, base_point=self.landmarks_a
-        )
-        expected = self.l2_metric_s2.dist(self.landmarks_a, self.landmarks_b) ** 2
-
-        self.assertAllClose(result, expected)
-
-    def test_l2_metric_log_and_exp(self):
```

Comment:
This test was passing, but the new test is skipped with:
`skip_test_exp_log_composition = True`
 
Please add this test back. (We do not want decrease in protection of the codebase).